dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -261,9 +256,65 @@ def _convert_units(self, df, project_metric_records, pivoted_columns):
                 break
 
         dataset_records = self._config.get_dimension(DimensionType.METRIC).get_records_dataframe()
-        df = energy.convert_units(
-            df, pivoted_columns, dataset_records, mapping_records, project_metric_records
+        match self._config.get_table_format_type():
+            case TableFormatType.PIVOTED:
+                df = energy.convert_units_pivoted(
+                    df, value_columns, dataset_records, mapping_records, project_metric_records
+                )
```

Comment:
This piece of code does not care and has always supported any dimension being pivoted. This is only mapping dataset dimensions to project dimensions. There has been another case that only supports metric. That is where the query asks to aggregate the pivoted dimension, such as with `end_uses_by_fuel_type`. In that example for StandardScenarios we are converting all energy types to MWh and so there is a second unit conversion that must occur. Up until now that code did not support other dimension types as the pivoted type. I tried to address that limitation in this PR (refer to pivoted_table.py). However, I did not remove the exception because it isn’t tested.